src/utils.rs

```diff
@@ -0,0 +1,14 @@
+use std::fs::File;
+use std::io::{BufWriter, Write};
+use std::path::Path;
+
+use anyhow::Result;
+use serde::Serialize;
+
+pub fn save_json_to_file<C: ?Sized + Serialize>(path: impl AsRef<Path>, content: &C) -> Result<()> {
```

Comment:
This module is not used. Please add it later when needed / mark it as public in main `lib.rs` / mark it as `#[allow(dead_code)]`